aii-opennebula/src/main/perl/opennebula.pm

```diff
@@ -64,18 +66,23 @@ sub make_one
     return $one;
 }
 
-sub process_template 
+# Detect and process ONE templates
+sub process_template
 {
     my ($self, $config, $tt_name) = @_;
-    my $res;
-    my $tt_rel = "metaconfig/opennebula/$tt_name.tt";
+    
     my $tree = $config->getElement('/')->getTree();
-    my $tpl = Template->new(INCLUDE_PATH => TEMPLATEPATH);
-    if (! $tpl->process($tt_rel, $tree, \$res)) {
-        $main::this_app->error("TT processing of $tt_rel failed: ", $tpl->error());
+    my $tpl = CAF::TextRender->new(
+        $tt_name,
+        $tree,
+        relpath => 'aii-opennebula',
+        log => $self,
+        );
+    if (!$tpl) {
```

Comment:
I just need a confirmation here. I assume `!$tpl` means new above returns an object, only that when stringified it becomes `""`, right?

Otherwise, the `$tpl->{fail}` below will be an attempt to dereference an undefined value, and crash.